Extension/src/LanguageServer/client.ts

```diff
@@ -1294,21 +1299,26 @@ export class DefaultClient implements Client {
         this.rootFolder = workspaceFolder;
         this.rootRealPath = this.RootPath ? (fs.existsSync(this.RootPath) ? fs.realpathSync(this.RootPath) : this.RootPath) : "";
 
-        let storagePath: string | undefined;
-        if (util.extensionContext) {
-            const path: string | undefined = util.extensionContext.storageUri?.fsPath;
-            if (path) {
-                storagePath = path;
-            }
-        }
+        this.cacheStoragePath = util.getCacheStoragePath();
```

Comment:
As alluded to in my last comment, can you move the setup of `cacheStoragePath` and `databaseStoragePath` into `createLanguageClient`, immediately before these strings are added to `initializationOptions`, since that is the only place these are used?